packages/signers/src/web3auth/signer.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  WalletClientSigner,
+  type SignTypedDataParams,
+  type SmartAccountAuthenticator,
+} from "@alchemy/aa-core";
+import { Web3Auth, type Web3AuthOptions } from "@web3auth/modal";
+import { createWalletClient, custom, type Hash } from "viem";
+import type {
+  Web3AuthAuthenticationParams,
+  Web3AuthUserInfo,
+} from "./types.js";
+
+export class Web3AuthSigner
+  implements
+    SmartAccountAuthenticator<
+      Web3AuthAuthenticationParams,
+      Web3AuthUserInfo,
+      Web3Auth
+    >
+{
+  inner: Web3Auth;
+  private signer: WalletClientSigner | undefined;
+
+  constructor(params: Web3AuthOptions | { inner: Web3Auth }) {
+    if ("inner" in params) {
+      this.inner = params.inner;
+      return;
+    }
+
+    this.inner = new Web3Auth(params);
+  }
+
+  get signerType() {
+    return "web3auth";
+  }
```

Comment:
why this? instead of just having
```
readonly signerType = "web3auth"
```

as a property. if we want to disable setting, we should probably have a `set signerType() { throw new Error("nope.") }`